motionplan/armplanning/plan_manager.go

```diff
@@ -347,29 +339,120 @@ func interp(start, end referenceframe.FrameSystemPoses, delta float64) reference
 	return mid
 }
 
-func (pm *planManager) foo(ctx context.Context, start referenceframe.FrameSystemInputs, goal referenceframe.FrameSystemPoses) error {
-	psc, err := newPlanSegmentContext(ctx, pm.pc, start, goal)
+func (pm *planManager) quickReroute(ctx context.Context,
+	psc *planSegmentContext,
+	goal referenceframe.FrameSystemInputs,
+) ([]referenceframe.FrameSystemInputs, error) {
+	goalPoses, err := goal.ComputePoses(pm.pc.fs)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	planSeed, err := initRRTSolutions(ctx, psc)
-	if err != nil {
-		return err
-	}
+	mid := interp(psc.startPoses, goalPoses, .5)
 
-	if planSeed.steps == nil {
-		return fmt.Errorf("no steps")
-	}
+	pm.logger.Infof("quickReroute\n\tstart: %v\n\tmid: %v\n\tgoal: %v", psc.startPoses, mid, goalPoses)
 
-	if len(planSeed.steps) != 1 {
-		return fmt.Errorf("steps odd %d", len(planSeed.steps))
-	}
+	numPoints := 4
 
-	err = psc.checkPath(ctx, start, planSeed.steps[0])
-	if err != nil {
-		return err
+	angleStep := 2 * math.Pi / float64(numPoints)
+	for j := .5; j <= 2; j += .5 {
+		for i := 0; i < numPoints; i++ {
+			angle := float64(i) * angleStep
+			attempt := referenceframe.FrameSystemPoses{}
+
+			offsetDistance := 0.0
+
+			for f, s := range psc.startPoses {
+				m := mid[f]
+
+				sPoint := s.Pose().Point()
+				mPoint := m.Pose().Point()
+				vec := mPoint.Sub(sPoint)
+
+				// Compute a perpendicular vector using cross product with a reference vector
```

Comment:
These comments helped me understand a lot, thanks! I've pushed a higher level description above the triple for-loop.